### ISSUE TITLE
[Bug] Voice icon on search widget crashes when google app disabled after widget installed

### ISSUE THREAD

**Issue by sblatz** [mozilla-mobile/fenix#11051]
## Steps to reproduce
1. Install the search widget with Google app enabled on your device
1. Disable the Google app.
1. Tap the "voice icon" on the search widget.

### Expected behavior
1. The voice icon should disappear as the functionality is no longer available

### Actual behavior
1. Fenix crashes when attempting to start the dictation.

[First mentioned here](https://github.com/mozilla-mobile/fenix/issues/10803#issuecomment-635813814).

**Comment by sblatz** [comment]
This feels like a pretty small edge case to me, honestly.

**Comment by BranescuMihai** [comment]
I would normally say that maybe we should redirect the user to settings, where they can re-enable the Google app. But that requires also some UI like a pop-up informing them of why this is happening, and then handling the result from settings. While this is certainly feasible, since the use-case is so limited, I think simply closing the activity OR redirecting to home and displaying a short Toast like "Please enable your Google App" should do. If that's the way to go, can we also get the string for the toast message?

**Comment by AmyYLee** [comment]
@brampitoyo @betsymi for follow-up

**Comment by AmyYLee** [comment]
@brampitoyo and @betsymi - Just following up on this. If this requires more attention, please update flag to needs UX-investigation and remove UX-feedback flag. Thanks!

**Comment by brampitoyo** [comment]
@sblatz @BranescuMihai

As the process to reactivate voice is too long and hard to teach with a snackbar that disappears after a few seconds, we should not show a toast.

Instead, let’s “scale down elegantly” and just hide the microphone icon from the widget, as we already do in the Fenix address bar.

**Comment by BranescuMihai** [comment]
@brampitoyo so the issue is that we're currently allowed by the system to update the widget only every hour. Which means that if the user disabled the Google app right after a scheduled widget update, they will see the microphone icon for another 59 minutes, even though it will do nothing on click. And only after those 59 minutes we'll be able to hide it.

Is that ok?

**Comment by brampitoyo** [comment]
@BranescuMihai I think showing the microphone icon for the next 59 minutes will be mildly annoying and could imply that something is broken (“I’ve disabled it – why does it still show up?), but it’s also the best compromise we can make given system limitation.

Let’s do that.

**Comment by Diana-Rus** [comment]
Hi issue is fixed, verification based on [comment](https://github.com/mozilla-mobile/fenix/issues/11051#issuecomment-671731789) on latest Nightly 8/26 with:
- Google Pixel 3 XL (Android 9)
- Samsung Galaxy S9 (Android 8)
- Sony Xperia Z5 (Android 7)
<description><summary>Gif</summary>

![20200826-095155](https://user-images.githubusercontent.com/49859863/91271002-2f16a780-e782-11ea-8491-332e9f16a7fd.gif)
</description>

@Mugurell - Based on the previous affirmation that the widget updates hourly. i can see that the mic icon is not removed from the widget, the update of it can't be seen on android device.
Created a new issue for this: https://github.com/mozilla-mobile/fenix/issues/14206

Note:
- In case of having disabled the google app and restarting the device will have the mic icon removed from the widget, even if the one hour timeline didn't passed
